client/context.go

```diff
@@ -0,0 +1,132 @@
+package client
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+)
+
+type ContextErrorResponse struct {
+	Status  int    `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+	Error   string `json:"error,omitempty"`
+}
+
+type ContextMetadata struct {
+	Name string `json:"name,omitempty"`
+}
+
+type Context struct {
+	Metadata ContextMetadata `json:"metadata,omitempty"`
+	Spec     ContextSpec     `json:"spec,omitempty"`
+	Version  string          `json:"version,omitempty"`
+}
+
+type ContextSpec struct {
+	Type string                 `json:"type,omitempty"`
+	Data map[string]interface{} `json:"data,omitempty"`
+}
+
+func (context *Context) GetID() string {
+	return context.Metadata.Name
+}
+
+func (client *Client) GetContext(name string) (*Context, error) {
+	fullPath := fmt.Sprintf("/contexts/%s?decrypt=true", url.QueryEscape(name))
```

Comment:
```suggestion
	fullPath := fmt.Sprintf("/contexts/%s?decrypt=true", url.PathEscape(name))
```
Using `QueryEscape` here produces odd results when `name` contains spaces. For example, `My Context/Other Piece' becomes `My+Context%2FOther+Piece`, which seems to return an HTTP 500 (confirmed via direct API call).

Using `PathEsacpe`, however, produces `My%20Context%2FOther%20Piece`, which _does_ return the context (and HTTP 200).